src/test/java/com/microsoft/sqlserver/jdbc/bulkCopy/bulkCopyTest.java

```diff
@@ -0,0 +1,181 @@
+import java.sql.Connection;
+import java.sql.Date;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.Time;
+import java.sql.Timestamp;
+
+import org.junit.jupiter.api.Test;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+
+import com.microsoft.sqlserver.jdbc.SQLServerBulkCopy;
+import com.microsoft.sqlserver.testframework.AbstractTest;
+import com.microsoft.sqlserver.testframework.DBConnection;
+import com.microsoft.sqlserver.testframework.DBResultSet;
+import com.microsoft.sqlserver.testframework.DBStatement;
+import com.microsoft.sqlserver.testframework.DBTable;
+import com.microsoft.sqlserver.testframework.sqlType.SqlBit;
+import com.microsoft.sqlserver.testframework.sqlType.SqlDateTime;;
+
+@RunWith(JUnitPlatform.class)
+public class bulkCopyTest extends AbstractTest{
+
+	/**
+	 * Test case for create table functionality {@link DBEngine#createTable(DBTable, com.microsoft.sqlserver.jdbc.SQLServerConnection)}
+	 * @throws Exception 
+	 */
+	@Test
+	public void testBulkCopyWithConnection() throws Exception {
+		
+		DBConnection con = new DBConnection(connectionString);
+		DBStatement stmt = con.createStatement();
+		DBTable sourceTable = new DBTable(true);
+		stmt.createTable(sourceTable);
+		stmt.populateTable(sourceTable);
+		
+		DBTable destinationTable = sourceTable.cloneSchema();
+		stmt.createTable(destinationTable);
+		DBResultSet srcResultSet = stmt.executeQuery("SELECT * FROM "+sourceTable.getEscapedTableName()+";");
+		SQLServerBulkCopy bulkCopy =new SQLServerBulkCopy((Connection)con.product());
+        bulkCopy.setDestinationTableName(destinationTable.getEscapedTableName());
+        bulkCopy.writeToServer((ResultSet)srcResultSet.product());
+        bulkCopy.close();
+        srcResultSet.close();
+        validateValues(con, sourceTable, destinationTable);
+        sourceTable.dropTable(stmt);
+        destinationTable.dropTable(stmt);
+	}
+	
+	
+	@Test
+	public void testBulkCopyWithConnectionString() throws Exception {
+		// Change Later
+		DBConnection con = new DBConnection(connectionString);
+		DBStatement stmt = con.createStatement();
+		DBTable sourceTable = new DBTable(false);
+		sourceTable.addColumn(new SqlBit());
+		sourceTable.addColumn(new SqlDateTime());
+		stmt.createTable(sourceTable);
+		stmt.populateTable(sourceTable);
+		DBTable destinationTable = sourceTable.cloneSchema();
+		stmt.createTable(destinationTable);
+		DBResultSet srcResultSet = stmt.executeQuery("SELECT * FROM "+sourceTable.getEscapedTableName()+";");
+		SQLServerBulkCopy bulkCopy =new SQLServerBulkCopy(connectionString);
+        bulkCopy.setDestinationTableName(destinationTable.getEscapedTableName());
+        bulkCopy.writeToServer((ResultSet)srcResultSet.product());
+        bulkCopy.close();
+        srcResultSet.close();
+        validateValues(con, sourceTable, destinationTable);
+        sourceTable.dropTable(stmt);
+        destinationTable.dropTable(stmt);
```

Comment:
can we move dropTable() methods into a finally block to make sure table is dropped when test fails.